physics/cu_unified_driver.F90

```diff
@@ -0,0 +1,1197 @@
+                             -(xmbs(i)*zus(i,k))
+                  trcflx_in1(k)=massflx(k)*.5*(clwtot+clwtot1)
+               endif
+             enddo
+
+             massflx   (1)=0.
+             trcflx_in1(1)=0.
+             call fct1d3 (kstop,kte,dtime_max,po_cup,                  &
+                            clw_in1,massflx,trcflx_in1,clw_ten(i,:),g)
+
+             do k=1,kstop
+               tem  = dt*(outqcs(i,k)*cutens(i)+outqc(i,k)*cuten(i)    &
+                      +outqcm(i,k)*cutenm(i)                           &
+                      +clw_ten(i,k)                                    &
+                         )
+               tem1 = max(0.0, min(1.0, (tcr-t(i,k))*tcrf))
+               if (clcw(i,k) .gt. -999.0) then
+                cliw(i,k) = max(0.,cliw(i,k) + tem * tem1)            ! ice
+                clcw(i,k) = max(0.,clcw(i,k) + tem *(1.0-tem1))       ! water
+               else
+                cliw(i,k) = max(0.,cliw(i,k) + tem)
+               endif
+
+             enddo
+
+            gdc(i,1,10)=forcing(i,1)
+            gdc(i,2,10)=forcing(i,2)
+            gdc(i,3,10)=forcing(i,3)
+            gdc(i,4,10)=forcing(i,4)
+            gdc(i,5,10)=forcing(i,5)
+            gdc(i,6,10)=forcing(i,6)
+            gdc(i,7,10)=forcing(i,7)
+            gdc(i,8,10)=forcing(i,8)
+            gdc(i,10,10)=xmb(i)
+            gdc(i,11,10)=xmbm(i)
+            gdc(i,12,10)=xmbs(i)
+            gdc(i,13,10)=hfx(i)
+            gdc(i,15,10)=qfx(i)
+            gdc(i,16,10)=pret(i)*3600.
+            if(ktop(i).gt.2 .and.pret(i).gt.0.)dt_mf(i,ktop(i)-1)=ud_mf(i,ktop(i))
+            endif
+            enddo
+!$acc end parallel
+!$acc kernels
+            do i=its,itf
+              if(pret(i).gt.0.)then
+                 cactiv(i)=1
+                 raincv(i)=.001*(cutenm(i)*pretm(i)+cutens(i)*prets(i)+cuten(i)*pret(i))*dt
+              else
+                 cactiv(i)=0
+                 if(pretm(i).gt.0)raincv(i)=.001*cutenm(i)*pretm(i)*dt
+              endif   ! pret > 0
+
+              if(pretm(i).gt.0)then
+                 cactiv_m(i)=1
+              else
+                 cactiv_m(i)=0
+              endif
+
+              ! Unify ccn
+              if(ccn_m(i).lt.ccn_gf(i))then
+                ccn_gf(i)=ccn_m(i)
+              endif
+
+              if(ccn_gf(i)<0) ccn_gf(i)=0
+
+              ! Convert ccn back to aod
+              aod_gf(i)=0.0027*(ccn_gf(i)**0.64)
+              if(aod_gf(i)<0.007)then
+                aod_gf(i)=0.007
+                ccn_gf(i)=(aod_gf(i)/0.0027)**(1/0.640)
+              elseif(aod_gf(i)>aodc0)then
+                aod_gf(i)=aodc0
+                ccn_gf(i)=(aod_gf(i)/0.0027)**(1/0.640)
+              endif
+            enddo
+!$acc end kernels
+ 100    continue
+!
+! Scale dry mixing ratios for water wapor and cloud water to specific humidy / moist mixing ratios
+!
+!$acc kernels
+        qv_spechum = qv/(1.0_kind_phys+qv)
```

Comment:
This kind of thing could be controlled by a flag to be more portable for other hosts.